glyphs2fontir/src/source.rs

```diff
@@ -545,105 +585,125 @@ impl Work<Context, WorkId, Error> for GlobalMetricWork {
 
             metrics.set_if_some(GlobalMetric::CapHeight, pos.clone(), master.cap_height());
             metrics.set_if_some(GlobalMetric::XHeight, pos.clone(), master.x_height());
-            metrics.set_if_some(
+            metrics.set_if_some_or_fallback(
                 GlobalMetric::Os2TypoAscender,
                 pos.clone(),
                 master.typo_ascender.map(|v| v as f64),
+                font.typo_ascender.map(|v| v as f64),
```

Comment:
okay so the other way to write all of these would be to stick with `set_if_some`, but then do,

```rust
metrics.set_if_some(
    MetricName,
    pos.clone(),
    master.typo_ascender.or(font.typo_ascender).map(|v| v as f64),
)
```

which might be... slightly better, on balance? at least it saves us adding the additional method. This whole big function is kind of calling out for a macro, but that's another story; let's just get it working first.

Reply:
Sounds good. I’ll try to do this in the morning.